src/ArduinoIoTCloud.h

```diff
@@ -53,6 +53,12 @@ enum ArduinoIoTConnectionStatus {
   IOT_STATUS_ERROR_GENERIC
 };
 
+enum ArduinoIoTSynchronizationStatus {
```

Comment:
I'm kind of an enum class guy ;) myself which is why I'd suggest using enum classes intead. The advantage of enum class over a plain-old-enum is
* type checking by compiler
* better readability due to the enum name being used with every enum constant.
```suggestion
enum class SyncStatus {
  Synchronized,
  WaitForCloud,
  ValuesProcessed
```
and the usage be like:
```C++
case SyncStatus::Synchronized: /* Do Stuff */ break;
```

Reply:
I like this!

Reply:
Done. @ubidefeo please change also your enums :-)